rfay/ddev
# fix: Traefik dynamic config middleware assignment

### How to use/test:
* create a ddev project with `ddev config`
* copy/rename `/project/traefik/.traefik-dynamic-config-extensions-template.yaml.example` to remove `.example` extension
* Likewise for ~/.ddev/traefik/static_config.plugins.yaml`
* `ddev start` 
* Check that routers and middleware are loading and working properly at http://localhost:10999/dashboard/#/
* Do the same for a 2nd project and check to see that separate middleware has been created per router
* Feel free to try defining other middlewares in both of those files, or changing the router assignments. Make sure to follow the same formatting, particularly with the `{{ .App.Name }}-` prefixes

### TODO

- [ ] Move the verbose comments in the template files to documentation, and generally write/improve the documentation
- [ ] I don't believe that existing projects will receive the `/project/traefik/.traefik-dynamic-config-extensions-template.yaml.example` file, as it is created with `ddev config` so that the file would be available prior to the first `ddev start`. Should that stay there, but also add some conditional logic to `traefik.go` to check for its existence, creating it if it doesn't exist?
